opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -123,7 +125,7 @@ namespace Opm {
 
 
         const UnitSystem& m_deckUnitSystem;
-        const ParseContext& m_parseContext;
+        const ParseContext m_parseContext;
```

Comment:
> i.e. the pointers are usually set to NULL

A way of marking them for non-deletion. It's 100% implementation defined and can vary between objects though.

> true. in the case at hand I don't think it would affect performance even if ParseContext was an extremely complex beast like Deck. On the other hand, if that happens for objects that are used in tight loops, you will have a performance impact. (which is what I mean with "bad habits".)

But if they're to be copied anyway you can't avoid this cost, regardless of reference passing. If a copy isn't needed then the argument should obviously be a reference.